### PR TITLE
document how to add a new target

### PR DESCRIPTION
**Pull request by esc**
This adds documentation to the README.rst on how to add a new target. It
is quite specific and I hope I have captured all the steps.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -93,6 +93,29 @@ Examples::
 
 Please see the output of ``./switchboard.py -h`` for more information.
 
+Adding a new Target
+===================
+
+In order to add a new target, you need to:
+
+* Implement the logic for the target, by subclassing
```
Comment (stuartarchibald) [inline README.rst:101]: Might want spell out where to put this new target, i.e. switchboard.py

File: README.rst [hunk 2]
```diff
@@ -93,6 +93,29 @@ Examples::
...
+  ``NumbaIntegrationTestTarget`` and filling in the blanks.
```
Comment (stuartarchibald) [inline README.rst]: Perhaps?
```suggestion
  ``NumbaIntegrationTestTarget`` and overloading necessary methods."
```

File: README.rst [hunk 3]
```diff
@@ -93,6 +93,29 @@ Examples::
...
+* Add an appropriate stanza in the CI configuration files such as
+  ``.circleci/config.yml`````.
```
Comment (stuartarchibald) [inline README.rst]: too many backticks.

**Comment by esc** [comment]
@stuartarchibald ready for the next review.

File: README.rst [hunk 4]
```diff
@@ -93,6 +93,30 @@ Examples::
...
 Please see the output of ``./switchboard.py -h`` for more information.
 
+Adding a new Target
+===================
+
+In order to add a new target, you need to:
+
+* Implement the logic for the target, by subclassing
+  ``NumbaIntegrationTestTarget`` in the file ``switchboard.py`` and overloading
+  necessary methods."
```
Comment (stuartarchibald) [inline README.rst]:
```suggestion
  necessary methods.
```

Comment (esc) [reply README.rst]: yeah, not sure where that came from

**Review by stuartarchibald — APPROVED** [review]
Thanks for the doc update and fixes.

**Pull request merged**